src/validator.js

```diff
@@ -1,5 +1,17 @@
 import freeze from 'deep-freeze';
```

Comment:
The same goes for `this.errors.hasOwnProperty(field)`. Refactoring would reduce the number of level of abstractions used in each function.